scripts/example_benchmark_dataset_full_workflow.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import pathlib
+
+from paint import PAINT_ROOT
+from paint.data import StacClient
+from paint.data.dataset import PaintCalibrationDataset
+from paint.data.dataset_splits import DatasetSplitter
+from paint.util import set_logger_config
+
+set_logger_config()
+
+
+if __name__ == "__main__":
+    """
+    This script demonstrates the full workflow for loading a ``torch.Dataset`` based on a given benchmark, specifically
+    it includes:
+    - Downloading the necessary metadata to generate the dataset splits.
+    - Generating the dataset splits.
+    - Creating a ``torch.Dataset`` based on these splits, if necessary, downloading the appropriate data.
+    """
+    # Read in arguments.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--metadata_input",
+        type=pathlib.Path,
+        help="File containing the metadata required to generate the dataset splits.",
+        default=f"{PAINT_ROOT}/metadata/calibration_metadata_all_heliostats.csv",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=pathlib.Path,
+        help="Root directory to save outputs.",
+        default=f"{PAINT_ROOT}/benchmarks",
+    )
+    parser.add_argument(
+        "--split_type",
+        type=str,
+        help="The split type to apply.",
+        default="azimuth",
+    )
+    parser.add_argument(
+        "--train_size",
+        type=int,
+        help="The size of the training split.",
+        default=10,
+    )
+    parser.add_argument(
+        "--val_size",
+        type=int,
+        help="The size of the validation split.",
+        default=30,
+    )
+    parser.add_argument(
+        "--remove_unused_data",
+        type=bool,
+        help="Whether to remove unused data or not.",
```

Comment:
What is unused data? Can we be more specific here?

Reply:
I have included a more informative description!